### PR TITLE
jobs: add config cleanup job

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Added a recurring task to remove configurations older than `JobSettings.CleanupConfigThresholdDays` days.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43119

#### Release Note

```release-note
Added a new config cleanup job to regularly remove outdated entries from the database.
```

### CODE REVIEW

**Comment by agnivade** [comment]
I don't think it's necessary to go to the level of creating a job for this. It's simple enough to have a recurring task in the app. Take a look at https://github.com/mattermost/mattermost-server/blob/c6dcf460b2e6f3eff6c68161153cfde378d8fed1/app/server.go#L1502-L1507. That's all we need.

Just a new config setting (similar to `CleanupJobsThresholdDays`) which by default is `-1`, which means rows are not cleaned up by default, to maintain original behavior. We always want new behavior to be opt-in, rather than opt-out.

And then look up that setting in the recurring task and cleanup rows older than that.

**Comment by isacikgoz** [comment]
>  It's simple enough to have a recurring task in the app. 

Totally skipped that, very good point 🤦 Anyway, updated accordingly.

File: app/server.go [hunk 1]
```diff
@@ -1580,6 +1590,17 @@ func doJobsCleanup(s *Server) {
 	}
 }
 
+func doConfigCleanup(s *Server) {
+	if *s.Config().JobSettings.CleanupConfigThresholdDays < 0 || strings.HasPrefix(s.ConfigStore().Store.String(), "file://") {
```
Comment (agnivade) [inline app/server.go:1594]: Maybe better to use `!config.IsDatabaseDSN(..)` ?

File: config/database.go [hunk 2]
```diff
@@ -416,3 +416,19 @@ func (ds *DatabaseStore) String() string {
 func (ds *DatabaseStore) Close() error {
 	return ds.db.Close()
 }
+
+// removes configurations from database if they are older than treshold days.
+func (ds *DatabaseStore) cleanUp(tresholdDays int) error {
```
Comment (agnivade) [inline config/database.go:421]: Simpler to pass the exact createAt in ms from the app layer. Then the query can remain same for both DBs.

```
dur := time.Duration(*s.Config().JobSettings.CleanupConfigThresholdDays) * time.Hour * 24
expiry := model.GetMillisForTime(time.Now().Add(-dur))
```

And then `DELETE FROM Configurations Where CreateAt < ?`

Comment (isacikgoz) [reply config/database.go:421]: Sure, why not 👍

File: config/database.go [hunk 3]
```diff
@@ -416,3 +416,12 @@ func (ds *DatabaseStore) String() string {
 func (ds *DatabaseStore) Close() error {
 	return ds.db.Close()
 }
+
+// removes configurations from database if they are older than treshold.
```
Comment (agnivade) [inline config/database.go:420]: minor grammar nit: threshold

**Review by agnivade — APPROVED** [review]
Nice work!

**Comment by agnivade** [comment]
@isacikgoz - Let's mention the config knob in the release notes. :)

**Pull request merged**